ui/components/src/TimeChart/TimeChart.tsx

```diff
@@ -0,0 +1,364 @@
+import { CanvasRenderer } from 'echarts/renderers';
+import { EChart, OnEventsType } from '../EChart';
+import { ChartHandleFocusOpts, ChartHandle, TimeChartSeriesMapping } from '../model/graph';
+import { useChartsTheme } from '../context/ChartsThemeProvider';
+import {
+  clearHighlightedSeries,
+  enableDataZoom,
+  getFormattedAxisLabel,
+  getYAxes,
+  restoreChart,
+  ZoomEventData,
+} from '../utils';
+import { CursorCoordinates, TimeChartTooltip, TooltipConfig } from '../TimeSeriesTooltip';
+import { useTimeZone } from '../context/TimeZoneProvider';
+
+use([
+  EChartsLineChart,
+  GridComponent,
+  DatasetComponent,
+  DataZoomComponent,
+  MarkAreaComponent,
+  MarkLineComponent,
+  MarkPointComponent,
+  TitleComponent,
+  ToolboxComponent,
+  TooltipComponent,
+  LegendComponent,
+  CanvasRenderer,
+]);
+
+export interface TimeChartProps {
+  height: number;
+  data: TimeSeries[];
+  seriesMapping: TimeChartSeriesMapping;
+  timeScale?: TimeScale;
+  yAxis?: YAXisComponentOption;
+  unit?: UnitOptions;
+  grid?: GridComponentOption;
+  tooltipConfig?: TooltipConfig;
+  noDataVariant?: 'chart' | 'message';
+  syncGroup?: string;
+  onDataZoom?: (e: ZoomEventData) => void;
+  onDoubleClick?: (e: MouseEvent) => void;
+  __experimentalEChartsOptionsOverride?: (options: EChartsCoreOption) => EChartsCoreOption;
+}
+
+export const TimeChart = forwardRef<ChartHandle, TimeChartProps>(function TimeChart(
+  {
+    height,
+    data,
+    seriesMapping,
+    timeScale: timeScaleProp,
+    yAxis,
+    unit,
+    grid,
+    tooltipConfig = { wrapLabels: true },
+    noDataVariant = 'message',
+    syncGroup,
+    onDataZoom,
+    onDoubleClick,
+    __experimentalEChartsOptionsOverride,
+  },
+  ref
+) {
+  const chartsTheme = useChartsTheme();
+  const chartRef = useRef<EChartsInstance>();
+  const [showTooltip, setShowTooltip] = useState<boolean>(true);
+  const [tooltipPinnedCoords, setTooltipPinnedCoords] = useState<CursorCoordinates | null>(null);
+  const [isDragging, setIsDragging] = useState(false);
+  const [startX, setStartX] = useState(0);
+  const { timeZone } = useTimeZone();
+  const totalSeries = data?.length ?? 0;
+
+  let timeScale: TimeScale;
+  if (timeScaleProp === undefined) {
+    const commonTimeScale = getCommonTimeScale(data);
+    if (commonTimeScale === undefined) {
+      // set default to past 5 years
+      const today = new Date();
+      const pastDate = new Date(today);
+      pastDate.setFullYear(today.getFullYear() - 5);
+      const todayMs = today.getTime();
+      const pastDateMs = pastDate.getTime();
+      timeScale = { startMs: pastDateMs, endMs: todayMs, stepMs: 1, rangeMs: todayMs - pastDateMs };
+    } else {
+      timeScale = commonTimeScale;
+    }
+  } else {
+    timeScale = timeScaleProp;
+  }
+
+  useImperativeHandle(
+    ref,
+    () => {
+      return {
+        highlightSeries({ name }: ChartHandleFocusOpts) {
+          if (!chartRef.current) {
+            // No chart. Do nothing.
```

Comment:
If you like it, you could copy the updated comments I saw in another file: "When chart is undefined, do not highlight series when hovering over legend"